include/picongpu/plugins/radiation/Radiation.hpp

```diff
@@ -664,396 +659,197 @@ namespace picongpu
-                        }
-                        else if(i == idLabels::Frequency) /* detector frequency record */
-                        {
-                            myListOfStr.push_back("None");
-                            myListOfStr.push_back("detector frequency index");
-                        }
-                        myListOfStr.push_back("None");
-
-                        // convert to splash format
-                        helper::GetSplashArrayOfString getSplashArrayOfString;
-                        helper::GetSplashArrayOfString::Result myArrOfStr;
-                        myArrOfStr = getSplashArrayOfString(myListOfStr);
-                        splash::ColTypeString ctSomeListOfStr(myArrOfStr.maxLen);
-
-                        hdf5DataFile.writeAttribute(
-                            currentStep,
-                            ctSomeListOfStr,
-                            (meshesPathName + meshRecordLabels(i)).c_str(),
-                            "axisLabels",
-                            1u, /* ndims: 1D array */
-                            splash::Dimensions(myListOfStr.size(), 0, 0), /* size of 1D array */
-                            &(myArrOfStr.buffers.at(0)));
-
-
-                        /* unitDimension */
-                        std::vector<float_64> unitDimension(traits::NUnitDimension, 0.0);
-                        if(i == idLabels::Amplitude) /* amplitude record */
-                        {
-                            /* units Joule seconds -> Length^2 * Time^-1 * Mass^1 */
-                            unitDimension[traits::SIBaseUnits::length] = 2.0;
-                            unitDimension[traits::SIBaseUnits::time] = -1.0;
-                            unitDimension[traits::SIBaseUnits::mass] = 1.0;
-                        }
-                        else if(i == idLabels::Detector) /* detector direction record */
-                        {
-                            /* units none */
-                        }
-                        else if(i == idLabels::Frequency) /* detector frequency record */
-                        {
-                            /* units 1./second -> Time^-1  */
-                            unitDimension[traits::SIBaseUnits::time] = -1.0;
-                        }
-                        hdf5DataFile.writeAttribute(
-                            currentStep,
-                            ctDouble,
-                            (meshesPathName + meshRecordLabels(i)).c_str(),
-                            "unitDimension",
-                            1u,
-                            splash::Dimensions(traits::NUnitDimension, 0, 0),
-                            &(*unitDimension.begin()));
-                    }
-                    /* end required openPMD attributes for meshes */
-                    /* end openPMD attributes */
-
-                    hdf5DataFile.close();
+                    openPMDdataFile.flush();
+                    openPMDdataFileIteration.close();
```

Comment:
Makes more sense to switch these two lines, I think.

Reply:
Okay.

Reply:
@PrometheusPi you missed to handle this comment